amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/oas/parser/types/InlineOasTypeParser.scala

```diff
@@ -725,6 +725,18 @@ case class InlineOasTypeParser(entryOrNode: YMapEntryLike,
     }
   }
 
+  private def generateUndefinedRequiredProperties(requiredFields: Map[String, YNode], shape: NodeShape): Unit = {
+    val presentProperties = shape.fields.fields().map(_.field.value.name).toSeq
+    requiredFields.keys
+      .filter(!presentProperties.contains(_))
+      .foreach(requiredProperty => {
+        shape
+          .withProperty(requiredProperty)
```

Comment:
I think that for this case both the PropertyShape and the AnyShape should be annotated as VirtualElements, not sure if your already checked this with pope.